ertenclave/ertenclave.go

```diff
@@ -85,28 +115,76 @@ func GetProductSealKey() (key, keyInfo []byte, err error) {
 func GetSealKey(keyInfo []byte) ([]byte, error) {
 	var keyBuffer *C.uint8_t
 	var keySize C.size_t
-	if res := C.oe_get_seal_key_v2((*C.uint8_t)(&keyInfo[0]), C.size_t(len(keyInfo)), &keyBuffer, &keySize); res != C.OE_OK {
+
+	res, _, errno := syscall.Syscall6(
+		SYS_get_seal_key,
+		uintptr(unsafe.Pointer(&keyInfo[0])),
+		uintptr(len(keyInfo)),
+		uintptr(unsafe.Pointer(&keyBuffer)),
+		uintptr(unsafe.Pointer(&keySize)),
+		0,
+		0,
+	)
+	if errno == syscall.ENOSYS {
+		return make([]byte, int(keySize)), errors.New("OE_UNSUPPORTED")
+	}
+	if errno != 0 {
+		return nil, fmt.Errorf("Error returned %d", errno)
+	}
+	if res != 0 {
 		return nil, oeError(res)
 	}
 
 	key := C.GoBytes(unsafe.Pointer(keyBuffer), C.int(keySize))
-	C.oe_free_seal_key(keyBuffer, nil)
+	_, _, errno = syscall.Syscall(
+		SYS_free_seal_key,
+		uintptr(unsafe.Pointer(keyBuffer)),
+		0,
+		0,
+	)
 	return key, nil
 }
 
-func getSealKeyByPolicy(sealPolicy C.oe_seal_policy_t) (key, keyInfo []byte, err error) {
+func getSealKeyByPolicy(sealPolicy uint32) (key, keyInfo []byte, err error) {
```

Comment:
```suggestion
func getSealKeyByPolicy(sealPolicy uintptr) (key, keyInfo []byte, err error) {
```
Does this work? Can we omit the cast then?